internal/attestation/slsa_provenance_02_test.go

```diff
@@ -0,0 +1,337 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package attestation
+
+import (
+	"bytes"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"testing"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/in-toto/in-toto-golang/in_toto"
+	v02 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+	ct "github.com/sigstore/cosign/pkg/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/hacbs-contract/ec-cli/internal/output"
+	e "github.com/hacbs-contract/ec-cli/pkg/error"
+)
+
+type mockLayer struct {
+	*mock.Mock
+}
+
+func (l mockLayer) Digest() (v1.Hash, error) {
+	args := l.Called()
+
+	return args.Get(0).(v1.Hash), args.Error(1)
+}
+
+func (l mockLayer) DiffID() (v1.Hash, error) {
+	args := l.Called()
+
+	return args.Get(0).(v1.Hash), args.Error(1)
+}
+
+func (l mockLayer) Compressed() (io.ReadCloser, error) {
+	args := l.Called()
+
+	return args.Get(0).(io.ReadCloser), args.Error(1)
+}
+
+func (l mockLayer) Uncompressed() (io.ReadCloser, error) {
+	args := l.Called()
+
+	return args.Get(0).(io.ReadCloser), args.Error(1)
+}
+
+func (l mockLayer) Size() (int64, error) {
+	args := l.Called()
+
+	return args.Get(0).(int64), args.Error(1)
+}
+
+func (l mockLayer) MediaType() (types.MediaType, error) {
+	args := l.Called()
+
+	return args.Get(0).(types.MediaType), args.Error(1)
+}
+
+func TestSLSAProvenanceFromLayer(t *testing.T) {
+	cases := []struct {
+		name      string
+		layer     v1.Layer
+		setup     func(l *mockLayer)
+		data      string
+		statement in_toto.ProvenanceStatementSLSA02
+		err       e.Error
+	}{
+		{
+			name: "nil layer",
+			err:  AT001,
+		},
+		{
+			name:  "media type error",
+			layer: &mockLayer{&mock.Mock{}},
```

Comment:
This is the same for all cases, maybe just have it done in the sub-test body? Same for `TestEntitySignature`.